### ISSUE TITLE
Performance issue retrieving OrientGraph through Connection Pool from other Thread

### ISSUE THREAD

**Issue by matthiasbe** [orientechnologies/orientdb#8443]
## Expected behavior  
`OrientGraphFactory.getTx()` and `OrientGraphFactory.getNoTx()` should take less than 1 ms as it was the case on our last used version 2.1.19.

## Actual behavior  
Executing `OrientGraphFactory.getTx()` or `OrientGraphFactory.getNoTx()` takes about 250ms on my local windows environment. It's the same results as if we were not using Connection Pool at all. It seems like new connection are never reused and are reopened everytime we ask for one.

## Steps to reproduce  
Our Connection Pool manager :
```java
public class OrientGraphConnection {

	private OrientGraphFactory instance;

	@PostConstruct
	public void init() {
		String location = dbSettings.getURI();
		log.info("Connection to " + location);

		instance = new OrientGraphFactory(location, dbSettings.getUsername(), dbSettings.getPasswd()).setupPool(100, 200);
		instance.setUseLog(false);
	}

	public OrientGraph getGraph() {
		return instance.getTx();
	}

	public OrientGraphNoTx getGraphNoTx() {
		return instance.getNoTx();
	}
}
```

Call for several `getGraph()` and `getGraphNoTx()` during treatment of a request, sequantially. Each time it takes about 250ms.

**Note** : the number of available connections seems ok, it doesn't increase when calling graphs sequentially, but it does when calling them in parallel.

**Extra information** : After some digging, we found out this latency is not present when calling `getGraph()` from *main Thread* and present when calling from *another Thread*.

**Comment by matthiasbe** [comment]
This seems to be fixed with release 3.0.5

**Comment by luigidellaquila** [comment]
Excellent, closing then

Thanks!

Luigi
